tests/tokenization/test_tokenization_utils.py

```diff
@@ -283,3 +283,38 @@ def test_instantiation_from_tokenizers_json_file(self):
         with tempfile.TemporaryDirectory() as tmpdirname:
             bert_tokenizer.save(os.path.join(tmpdirname, "tokenizer.json"))
             PreTrainedTokenizerFast(tokenizer_file=os.path.join(tmpdirname, "tokenizer.json"))
+
+    @require_tokenizers
+    def test_split_special_tokens(self):
+        tokenizer = PreTrainedTokenizerFast.from_pretrained('bert-base-cased', split_special_tokens=True)
+        text = 'Here is an example of bos token: [CLS]'
+
+        encoded_unsplit_val = {'input_ids': [101, 3446, 1110, 1126, 1859, 1104, 171, 2155, 22559, 131, 101, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
+        encoded_split_val = {'input_ids': [101, 3446, 1110, 1126, 1859, 1104, 171, 2155, 22559, 131, 164, 140, 15928, 166, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
```

Comment:
```suggestion
        encoded_unsplit_val = {'input_ids': [101, 3446, 1110, 1126, 1859, 1104, 171, 2155, 22559, 131, 101, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]} # fmt: skip
        encoded_split_val = {'input_ids': [101, 3446, 1110, 1126, 1859, 1104, 171, 2155, 22559, 131, 164, 140, 15928, 166, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]} # fmt: skip
```
you will probably need that to prevent make fixup from splitting this in many lines